patches/net/minecraft/client/multiplayer/MultiPlayerGameMode.java.patch

```diff
@@ -8,7 +8,7 @@
                      block.playerWillDestroy(level, p_105268_, blockstate, this.minecraft.player);
                      FluidState fluidstate = level.getFluidState(p_105268_);
 -                    boolean flag = level.setBlock(p_105268_, fluidstate.createLegacyBlock(), 11);
-+                    boolean flag = blockstate.onDestroyedByPlayer(level, p_105268_, minecraft.player, false, fluidstate);
++                    boolean flag = blockstate.onDestroyedByPlayer(level, p_105268_, minecraft.player, minecraft.player.getMainHandItem(), false, fluidstate);
```

Comment:
Should we always be copying this stack? It would ensure against silently modifying the main hand item through the parameter and it would be consistent with the idea of taking a snapshot of the item (as is done in the file below so that the item is in the pre-damage state)

Reply:
Done for both single-player (commented here), and multiplayer creative.

Should we defensive clone for the smp call also? It's already a copy of the main hand, but gets used down the line for `playerDestroy` and `PlayerDestroyItemEvent`.